rules/integrations/aws_bedrock/aws_bedrock_multiple_validation_exception_errors_by_single_user.toml

```diff
@@ -0,0 +1,58 @@
+[metadata]
+creation_date = "2024/09/11"
+integration = ["aws_bedrock"]
+maturity = "production"
+updated_date = "2024/09/11"
+min_stack_comments = "ES|QL rule type is still in technical preview as of 8.13, however this rule was tested successfully; integration in tech preview"
+min_stack_version = "8.13.0"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies multiple validation exeception errors within AWS Bedrock. Validation errors occur when you run the InvokeModel or 
+InvokeModelWithResponseStream APIs on a foundation model that uses an incorrect inference parameter or corresponding value. 
+These errors also occur when you use an inference parameter for one model with a model that doesn't have the same API parameter.
+This could indicated attempts to bypass limitations of other approved models, or to force an impact on the environment by incurring
+exhorbitant costs.
+"""
+false_positives = ["Legitimate misunderstanding by users or overly strict policies"]
+from = "now-60m"
+interval = "10m"
+language = "esql"
+license = "Elastic License v2"
+name = "AWS Bedrock Detected Multiple Validation Exception Errors by a Single User"
+references = [
+    "https://atlas.mitre.org/techniques/AML.T0015",
+    "https://atlas.mitre.org/techniques/AML.T0034",
+    "https://atlas.mitre.org/techniques/AML.T0046",
+    "https://www.elastic.co/security-labs/elastic-advances-llm-security"
+]
+risk_score = 73
+rule_id = "725a048a-88c5-4fc7-8677-a44fc0031822"
+setup = """## Setup
+
+This rule requires that AWS Bedrock Integration be configured. For more information, see the AWS Bedrock integration documentation:
+
+https://www.elastic.co/docs/current/integrations/aws_bedrock
+"""
+severity = "high"
+tags = [
+    "Domain: LLM",
+    "Data Source: AWS Bedrock",
+    "Data Source: AWS S3",
+    "Resources: Investigation Guide",
+    "Use Case: Policy Violation",
+    "Mitre Atlas: T0015",
+    "Mitre Atlas: T0034",
+    "Mitre Atlas: T0046",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-aws_bedrock.invocation-*
+| where gen_ai.response.error_code == "ValidationException"
```

Comment:
If I understand correctly, we are looking for atleast 3 instances of `ValidationException` in the last hour for a single user and model? My concern is dev or testing false-positives. In a emulation or attack, is it plausible that these happen in a smaller time window that is less likely the result of dev or testing?

If so, we can truncate timestamps and count how many occurrences happen in that window within the last hour. Here's an example:

```
from logs-aws_bedrock.invocation-*

// truncate the timestamp to a 1-minute window
| eval target_time_window = DATE_TRUNC(1 minutes, @timestamp)

| where gen_ai.response.error_code == "ValidationException"

// count the number of users causing validation errors within a 1 minute window
| stats total_denials = count(*)  target_time_window, user.id,  cloud.account.id
| where total_denials > 3
```